test/e2e/specs/editor/plugins/block-directory.spec.js

```diff
@@ -0,0 +1,258 @@
+		 * Be super weird so there won't be a matching block installed.
+		 */
+		await blockLibrary
+			.getByRole( 'searchbox', {
+				name: 'Search for blocks and patterns',
+			} )
+			.fill( '@$@@Dsdsdfw2$@' );
+
+		/**
+		 * The inserter can flash "No results found" before requesting results from the directory,
+		 * fulfilling the first assertion.
+		 * Waiting for these two elements ensures that the request was dispatched.
+		 */
+		await expect(
+			blockLibrary.locator( '.block-editor-inserter__no-results' )
+		).toHaveText( 'No results found.' );
+		await expect(
+			blockLibrary.locator( '.block-editor-inserter__tips' )
+		).toContainText( 'Interested in creating your own block?' );
+	} );
+
+	test( 'Should be able to add (the first) block', async ( {
+		editor,
+		page,
+	} ) => {
+		// Mock response for search with the block.
+		await page.route(
+			( url ) => matchUrl( url, SEARCH_URLS ),
+			async ( route, request ) => {
+				if ( request.method() === 'GET' ) {
+					await route.fulfill( {
+						json: [ MOCK_BLOCK1, MOCK_BLOCK2 ],
+					} );
+				} else {
+					await route.continue();
+				}
+			}
+		);
+
+		// Mock response for block type.
+		await page.route(
+			( url ) => matchUrl( url, BLOCK_TYPE_URLS ),
+			async ( route ) => {
+				await route.fulfill( {
+					json: {},
+				} );
+			}
+		);
+
+		// Mock response for install.
+		await page.route(
+			( url ) => matchUrl( url, INSTALL_URLS ),
+			async ( route ) => {
+				await route.fulfill( {
+					json: MOCK_INSTALLED_BLOCK_PLUGIN_DETAILS,
+				} );
+			}
+		);
+
+		// Mock the response for the js asset once it gets injected.
+		await page.route(
+			( url ) => url.href.includes( MOCK_BLOCK1.assets[ 0 ] ),
+			async ( route ) => {
+				await route.fulfill( {
+					contentType: 'application/javascript; charset=utf-8',
+					body: Buffer.from( block, 'utf8' ),
+				} );
+			}
+		);
+
+		// Mock the post-new page as requested via apiFetch for determining new CSS/JS assets.
+		await page.route(
+			( url ) => url.href.includes( '/post-new.php' ),
+			async ( route ) => {
+				await route.fulfill( {
+					contentType: 'text/html; charset=UTF-8',
+					body: `<html><head><script id="mock-block-js" src="${ MOCK_BLOCK1.assets[ 0 ] }"></script></head><body/></html>`,
+				} );
+			}
+		);
+
+		await page
+			.getByRole( 'toolbar', { name: 'Document tools' } )
+			.getByRole( 'button', { name: 'Toggle block inserter' } )
+			.click();
+
+		const blockLibrary = page.getByRole( 'region', {
+			name: 'Block Library',
+		} );
+		const downloadableBlock = page
+			.getByRole( 'listbox', {
+				name: 'Blocks available for install',
+			} )
+			.getByRole( 'option', {
+				name: `Install ${ MOCK_BLOCK1.title }.`,
+				exact: true,
+			} );
+
+		await blockLibrary
+			.getByRole( 'searchbox', {
+				name: 'Search for blocks and patterns',
+			} )
+			.fill( MOCK_BLOCK1.title );
+
+		await expect( downloadableBlock ).toBeVisible();
+
+		// Install the block.
+		await downloadableBlock.click();
```

Comment:
Since clicking the element implies visibility, can we skip the visibility check?

Reply:
Don't have a strong opinion here. It is just a personal translation for an action sequence - search, confirm the item is visible, and click on it 😄

Btw, our guide also suggests explicit assertions - https://developer.wordpress.org/block-editor/contributors/code/testing-overview/e2e/#make-explicit-assertions

cc @kevin940726

Reply:
If the visibility is part of what we want to test then I'd say explicit assertion is often better. In cases where it fails then we get a nicer error message on that line rather than an implicit error when trying to click on it. On the other hand, if checking the visibility isn't the focus of the test (maybe some other tests did it), then we can skip the visibility check 👍.